### PR TITLE
Do not query data that is already in the appconfig

### PR DESCRIPTION
**Pull request by rullzer**
This is already available. We better use a simple regex.

@LukasReschke as discussed!
CC: @blizzz

### CODE REVIEW

**Comment by rullzer** [comment]
If we agree here I'll also fix the other helper functions. Because they are very similar.

File: apps/user_ldap/lib/Helper.php [hunk 1]
```diff
@@ -55,32 +55,21 @@ class Helper {
...
-				AND `configkey` LIKE ?
-		';
-
-		if($activeConfigurations) {
-			if (\OC::$server->getConfig()->getSystemValue( 'dbtype', 'sqlite' ) === 'oci') {
-				//FIXME oracle hack: need to explicitly cast CLOB to CHAR for comparison
-				$sql .= ' AND to_char(`configvalue`)=\'1\'';
-			} else {
-				$sql .= ' AND `configvalue` = \'1\'';
+		$config = \OC::$server->getConfig();
```
Comment (LukasReschke) [inline apps/user_ldap/lib/Helper.php]: If we adjust the others as well we should change that to use DI and also have tests then :)

**Review by LukasReschke — APPROVED** [review]
I do like the changes but would appreciate DI and unit tests in the final version :)

File: apps/user_ldap/lib/Helper.php (hunk 1 above)
Comment (rullzer) [inline apps/user_ldap/lib/Helper.php]: Agreed. Testing would be good!

**Comment by MorrisJobke** [comment]
@LukasReschke @rullzer Should this get reviewed or will there be some additional stuff for DI and tests?

**Comment by rullzer** [comment]
@MorrisJobke yeah let me just finish this today with tests DI and the works. Better do it right right away.

**Comment by rullzer** [comment]
OK review time!

**Comment by MorrisJobke** [comment]
Tested and works 👍

**Review by MorrisJobke — APPROVED** [review]

**Comment by LukasReschke** [comment]
LGTM

**Pull request merged**